### DISCUSSION TITLE
Beta 1.0.1.13

### DISCUSSION THREAD

**Discussion by Erabior** [Erabior/RouteManager#39] [General]
Reverted station selection logic to how it is handled in 1.0.1.3. This should fix issues #33 and #34.

<hr /><em>This discussion was created from the release <a href='https://github.com/Erabior/RouteManager/releases/tag/1.0.1.13'>Beta 1.0.1.13</a>.</em>

**Comment by AdamC1228** [comment]
Hypothetically we have a route that travels From Bryson -> ... -> Silva 

Lets say we pickup at Ella and there are 10 passengers for Bryson. Is it better to pickup those 10 passengers now and take them to the end of the line so that by the time we return to Ella, more passengers will have generated? And likewise for the rest of the stops? Or is it better to get to the end of the line and have a train of passengers destined only for the end station. 

I have not studied /analyzed the station behavior closely yet to know if this "Model" breaks down over time. It would appear the game has a waiting capacity defined for stations of 50 people that can potentially creep as high as 100 as time progresses. 

I can see where a small train of maybe 2 coaches would be saturated on capacity in this scenario with so many station selected but realistically if you have that amount of stations unlocked **and** are running a train between that many stations your coach count would not be 2 coaches either. 

Waiting Capacity:
![image](https://github.com/Erabior/RouteManager/assets/13957003/bb475907-642f-444c-81e2-ee7d5ca4e010)

Growth interval:
![image](https://github.com/Erabior/RouteManager/assets/13957003/9c297019-3305-4339-838d-12b340a3234d)

Point is, I don't know which is better at the moment but I welcome a discussion on the matter.

**Reply by nitroSkirata** [reply]
In the rush hours any train will propably be overloaded if you take any passenger onboard. Especially since there is no prioritized boarding for passengers travelling in the correct direction.

In the nighttime however, taking everyone onboard would lead to a higher utilization of the train.

In an ideal gamified world, Passengers would board by direction and distance and if the train is full passengers for the wrong direction will be offloaded to make space for those going in the right direction.

I will take a look at the occupation of the stations once day breaks in my save, but I am pretty sure I have seen as many as 130 Pax waiting at a station

**Reply by AdamC1228** [reply]
I have not noticed the passenger count get that high yet at any individual station. I will have to delve deeper into the code and figure out how the game calculates the waiting capacity then. 

I did notice that the game has predefined levels for each hour and they are defined as such:
![image](https://github.com/Erabior/RouteManager/assets/13957003/469ce51a-9193-43f7-bc1e-845086f91b71)

Peak hours of passengers seem to be 6am - 9am & 5pm - 6pm assuming midnight is the first hour.

**Reply by AdamC1228** [reply]
I went back and indeed there is more to the max waiting passengers than I initially noticed so yeah, if we did pursue this it would need to be at off-peak hours.

**Reply by Narflar** [reply]
with that in mind, what about customizable time-tables?  Even to the point of having an additional train assigned only to run during the peak hours or an express train.

**Reply by AdamC1228** [reply]
Time tables may be theoretically possible in a simplistic sense such as: Proceed from Bryson to Ella. Halt until x time ten proceed to Whittier halt until x time. 

Practically speaking there are several factors that would need to be considered when operating multiple trains servicing the same stations such as:

- For stations where the platform is single track (Looking at you Ella) the train would be blocking the line. Thus blocking all traffic until the train is moved manually or the timetable kicks in.
    - We could move the train to a siding but the mod at present does not have any capability of determining or throwing switches nor does it have a concept of signalling. Lot of code would need to be written.
- Having multiple trains serve the same station sets would require them to be able to pass or move out of the way / block. How would this be handled?

**Reply by Narflar** [reply]
I would expect the AI to work the same as normal.   If you don't have
signals then you have to manage everything as you would when running cargo
and passenger trains together.

Running multiple passenger trains and cargo would only make sense if you
have signals.  Then you don't have to worry about trains fighting for a
station or blocking.
